src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -537,12 +538,29 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
 
             try
             {
-                if (!process.WaitForExit((int)(ExecTimeout.TotalMilliseconds)))
+                var output = new StringBuilder();
+                process.OutputDataReceived += (_, args) =>
+                {
+                    if (args.Data != null)
+                    {
+                        output.Append(args.Data);
+                    }
+                };
+                process.BeginOutputReadLine();
+
+                if (!process.WaitForExit((int)ExecTimeout.TotalMilliseconds))
                 {
                     throw new KubeConfigException("external exec failed due to timeout");
                 }
 
-                var responseObject = KubernetesJson.Deserialize<ExecCredentialResponse>(process.StandardOutput.ReadToEnd());
+                // Force flush the output buffer to avoid case of missing data
+                if (ExecTimeout != Timeout.InfiniteTimeSpan)
+                {
+                    process.WaitForExit();
```

Comment:
any detail here?
got confused? why wait again

Reply:
This is necessary when `WaitForExit(Int32)` has been called with async event handlers - an additional call to `WaitForExit()` overload is required to ensure all the processing has completed else output data can be incomplete.

See:  https://github.com/dotnet/runtime/issues/18789#issuecomment-265359492
and https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=netstandard-2.0#system-diagnostics-process-waitforexit

>This overload ensures that all processing has been completed, including the handling of asynchronous events for redirected standard output. You should use this overload after a call to the [WaitForExit(Int32)](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=netstandard-2.0#system-diagnostics-process-waitforexit(system-int32)) overload when standard output has been redirected to asynchronous event handlers.